### PR TITLE
Add more skeleton code for Task validation.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

**Comment by dlorenc** [comment]
Ref #29

**Comment by bobcatfish** [comment]
Thanks for adding this @dlorenc !

Could you add [a bit more info to your commit message about some of the motivation for these changes](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#commit-messages), e.g. stuff like:

* What are these `SetDefaults` functions about and why do we need them?
* What will be calling `Validate` (I assume it's some kind of magic in `webhook.AdmissionController`)

And/or anything else you can think of to help future explorers

File: pkg/apis/pipeline/v1alpha1/task_defaults.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+*/
+
+package v1alpha1
+
+func (t *Task) SetDefaults() {
+	t.Spec.SetDefaults()
+}
+
+func (ts *TaskSpec) SetDefaults() {
+	ts.SetDefaults()
```
Comment (ImJasonH) [inline pkg/apis/pipeline/v1alpha1/task_defaults.go]: This is going to infinitely recurse, eventually opening a wormhole that will swallow the earth. Or, stack overflow, whichever is first.

Comment (bobcatfish) [reply pkg/apis/pipeline/v1alpha1/task_defaults.go]: image

**Comment by dlorenc** [comment]
Thanks @bobcatfish, added more explanation in the commit message. I also added one of those type assertions to make it clear what interface I was intending to implement.

**Comment by bobcatfish** [comment]
thanks for updating the commit message!

/lgtm
/meow space

**Pull request merged**